File: src/sugtao4423/lod/main_fragment/MainFragmentPagerAdapter.java
package sugtao4423.lod.main_fragment;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import sugtao4423.lod.App;
import sugtao4423.lod.dataclass.TwitterList;

public class MainFragmentPagerAdapter extends FragmentStatePagerAdapter{

	private Context context;
	private App app;

	private Fragment_mention fragmentMention;
	private Fragment_home fragmentHome;

	public MainFragmentPagerAdapter(FragmentManager fm, Context context){
		super(fm);
		this.context = context;
		this.app = (App)context.getApplicationContext();

		fragmentMention = new Fragment_mention();
		fragmentHome = new Fragment_home();
	}

	@Override
	public Fragment getItem(int i){
		switch(i){
		case 0:
			return fragmentMention;
		case 1:
			return fragmentHome;
		default:
			return new Fragment_List(i - 2);
		}
	}

	@Override
	public int getCount(){
		return app.getLists(context).length + 2;
	}

	@Override
	public CharSequence getPageTitle(int position){
		TwitterList[] lists = app.getLists(context);
		switch(position){
		case 0:
			return "Mention";
		case 1:
			return "Home - Lv." + app.getLevel().getLevel();
		default:
			return lists[position - 2].getListName();
		}
	}

	public Fragment_mention getFragmentMention(){
		return fragmentMention;
	}

	public Fragment_home getFragmentHome(){
		return fragmentHome;
	}
}

File: app/src/main/java/sugtao4423/lod/main_fragment/MainFragmentPagerAdapter.kt
package sugtao4423.lod.main_fragment

import android.content.Context
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import sugtao4423.lod.App
import sugtao4423.lod.R

class MainFragmentPagerAdapter(fm: FragmentManager, private val context: Context) : FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    private val app = context.applicationContext as App
    val fragmentMention = Fragment_mention()
    val fragmentHome = Fragment_home()

    override fun getItem(i: Int): Fragment {
        return when (i) {
            0 -> fragmentMention
            1 -> fragmentHome
            else -> {
                val list = Fragment_List()
                val args = Bundle()
                args.putInt(Fragment_List.LIST_INDEX, i - 2)
                list.arguments = args
                return list
            }
        }
    }

    override fun getCount(): Int {
        return app.getLists(context).size + 2
    }

    override fun getPageTitle(position: Int): CharSequence {
        val lists = app.getLists(context)
        return when (position) {
            0 -> context.getString(R.string.page_title_mention)
            1 -> context.getString(R.string.param_page_title_home, app.getLevel().getLevel())
            else -> lists[position - 2].listName
        }
    }

}